src/test/run-make/json-errors/Makefile

```diff
@@ -0,0 +1,8 @@
+-include ../tools.mk
+
+all:
+	cp foo.rs $(TMPDIR)
+	cd $(TMPDIR)
+	$(RUSTC) -Z unstable-options --error-format=json foo.rs 2>foo.log || true
+	grep -q '{"message":"unresolved name `y`","code":{"code":"E0425","explanation":"\\nAn unresolved name was used. Example of erroneous codes.*"},"level":"error","span":{"file_name":"foo.rs","byte_start":523,"byte_end":524,"line_start":14,"line_end":14,"column_start":18,"column_end":19},"children":\[\]}' foo.log
```

Comment:
I think it is a good idea - I imagine tools may want to present these explanations to users in some form, and this makes it easy. I don't think the size of the output matters at all - it just gets processed by tools, it's not meant to be human-readable. I don't think that it will get large enough to be a speed issue, even with explanations for hundreds of issues, there's still only a few 10s of Ks of data